### PR TITLE
support centralized process changes

### PR DESCRIPTION
**Pull request by jeffkaufman**

### CODE REVIEW

**Comment by oschaaf** [comment]
No comments, LGTM

File: src/ngx_rewrite_driver_factory.cc [hunk 1]
```diff
@@ -250,4 +250,23 @@ void NgxRewriteDriverFactory::InitStats(Statistics* statistics) {
...
 
+void NgxRewriteDriverFactory::PrepareForkedProcess(const char* name) {
+  ngx_pid = ngx_getpid();  // Needed for logging to have the right PIDs.
+  SystemRewriteDriverFactory::PrepareForkedProcess(name);
+}
+
+void NgxRewriteDriverFactory::NameProcess(const char* name) {
+  SystemRewriteDriverFactory::NameProcess(name);
+
+  // Superclass set status with prctl.  Nginx has a helper function for setting
```
Comment (hillsp) [inline src/ngx_rewrite_driver_factory.cc]: I don't quite follow this. Why do you want to change the name *twice*?

File: test/nginx_system_test.sh [hunk 2]
```diff
@@ -577,6 +577,18 @@ check test $(scrape_stat image_rewrite_total_original_bytes) -ge 10000
 # happens both before and after.
 start_test "Reload config"
 
+function check_process_names() {
+  if ! $USE_VALGRIND; then
+    # There should be one babysitter and controller running.  Under valgrind
+    # process labels are confused, so skip the check then.
+
+    check [ $(ps aux | grep -c 'nginx: pagespeed babysitte[r]') -eq 1 ]
```
Comment (hillsp) [inline test/nginx_system_test.sh]: pgrep -c would be nice here, if it were installed.

Comment (jeffkaufman) [inline test/nginx_system_test.sh]: Yes, but I don't want to add a pgrep dependency.

File: src/ngx_rewrite_driver_factory.cc (hunk 1 above)
Comment (jeffkaufman) [inline src/ngx_rewrite_driver_factory.cc]: There are two kinds of process "names": there's the status, 16 chars you can set with `prctl`, and there's `argv[0]`.  Some ways of looking at processes use one, some use the other.  Nginx processes do typically set `argv[0]` so I'd like to do that to fit in.  That's awkward enough [1] that I only want to do it in situations where a helper function is available.

[1] https://cs.corp.google.com/piper///depot/google3/third_party/nginx/src/src/os/unix/ngx_setproctitle.c?q=ngx_setproctitle&l=79&dr=CSs

**Comment by hillsp** [comment]
LGTM

**Comment by jeffkaufman** [comment]
This went in as 8efb831b

**Pull request closed without merging**